server/test/api/inventory.js

```diff
@@ -0,0 +1,202 @@
+  });
+
+  // ========================== inventory groups ==============================
+
+  // create inventory group
+  it('POST /inventory/group create a new inventory group', () => {
+    return agent.post('/inventory/groups')
+      .send(inventoryGroup)
+      .then((res) => {
+        helpers.api.created(res);
+        expect(res.body.uuid).to.be.equal(inventoryGroup.uuid);
+      })
+      .catch(helpers.handler);
+  });
+
+  // update inventory group
+  it('PUT /inventory/group/:uuid updates an existing inventory group', () => {
+    return agent.put('/inventory/groups/' + inventoryGroup.uuid)
+      .send(updateGroup)
+      .then((res) => {
+        let group = res.body[0];
+        updateGroup.uuid = inventoryGroup.uuid;
+        expect(group).to.contain.all.keys(Object.keys(updateGroup));
+        for(var i in group) {
+          expect(group[i]).to.be.equals(updateGroup[i]);
+        }
+      })
+      .catch(helpers.handler);
+  });
+
+  // list of inventory groups
+  it('GET /inventory/group returns list of inventory groups', () => {
+    return agent.get('/inventory/groups')
+      .then((res) => {
+        countInventoryGroups = res.body.length;
+        helpers.api.listed(res, countInventoryGroups);
+      })
+      .catch(helpers.handler);
+  });
+
+  // detailS of inventory groups
+  it('GET /inventory/group returns details of an inventory group', () => {
+    return agent.get('/inventory/groups/' + inventoryGroup.uuid)
+      .then((res) => {
+        let group = res.body[0];
+        expect(group).to.contain.all.keys(Object.keys(inventoryGroup));
+        // compare value to the last update of our request
+        for(var i in group) {
+          expect(group[i]).to.be.equals(updateGroup[i]);
+        }
+        helpers.api.listed(res, 1);
+      })
+      .catch(helpers.handler);
+  });
+
+  // ========================== inventory types ===============================
+
+  // create inventory type
+  it('POST /inventory/types create a new inventory types', () => {
+    return agent.post('/inventory/types')
+      .send(inventoryType)
+      .then((res) => {
+        inventoryType.id = res.body.id;
+        helpers.api.created(res);
+        expect(res.body.id).to.be.equal(inventoryType.id);
+      })
+      .catch(helpers.handler);
+  });
+
+  // update inventory type
+  it('PUT /inventory/types/:id updates an existing inventory type', () => {
+    return agent.put('/inventory/types/' + inventoryType.id)
+      .send(updateType)
+      .then((res) => {
+        let type = res.body[0];
+        updateType.id = inventoryType.id;
+        expect(type).to.contain.all.keys(Object.keys(updateType));
+        expect(type).to.be.deep.equals(updateType);
+      })
+      .catch(helpers.handler);
+  });
+
+  // list of inventory type
+  it('GET /inventory/types returns list of inventory types', () => {
+    return agent.get('/inventory/types')
+      .then((res) => {
+        countInventoryTypes = res.body.length;
```

Comment:
A stronger test would be:
```js
const countInventoryTypes = 3;
helpers.api.listed(res, countInventoryTypes):
```
With this test, you do not know if the number in the database changed between tests.

It is true that updating tests when the database changes is annoying, but the benefit is that it alerts the programmer that things changed.